tutorials/scripting/gdextension/gdextension_c_example.rst

```diff
@@ -0,0 +1,2165 @@
+        api.object_set_instance_binding(object, class_library, &gdexample_class_binding_callbacks);
+        destructors.string_name_destructor(&class_name);
+
+        return object;
+    }
+
+    void gdexample_free_instance(void *p_class_userdata, GDExtensionClassInstancePtr p_instance)
+    {
+        if (p_instance == NULL)
+        {
+            return;
+        }
+        GDExample *self = (GDExample *)p_instance;
+        gdexample_class_destructor(self);
+        api.mem_free(self);
+    }
+
+When instantiating an object, first we create a new Sprite2D object, since
+that's the parent of our class. Then we allocate memory for our custom struct
+and call its constructor. We save the pointer to the Godot object in the struct
+as well like we mentioned earlier.
+
+Then we set our custom struct as the instance data. This will make Godot know
+that the object is an instance of our custom class and properly call our custom
+methods for instance, as well as passing this data back.
+
+Note that we return the Godot object we created, not our custom struct.
+
+For the ``gdextension_free_instance`` function, we just call the destructor and free the memory we
+allocated for the custom data. It is not necessary to destruct the Godot object
+since that will be taken care of by the engine itself.
+
+A demo project
+--------------
+
+Now that we can create and free our custom object, we should be able to try it
+out in an actual project. For this, you need to open Godot and create a new
+project on the ``demo`` folder. The project manager may warn you the folder
+isn't empty if you have compiled the extension before, you can safely ignore
+this warning this time.
+
+If you didn't compile the extension yet, it is the time to do it now. To do
+that, open a terminal or command prompt, navigate to the root folder of the
+extension and run ``scons``. It should compile quickly since the extension is
+very simple.
+
+Then, create a file called ``gdexample.gdextension`` inside the ``demo`` folder.
+This is a Godot resource that describes the extension, allowing the engine to
+properly load it. Put the following content in this file:
+
+.. code-block::
+
+    [configuration]
+
+    entry_symbol = "gdexample_library_init"
+    compatibility_minimum = "4.2"
+
+    [libraries]
+    macos.debug = "res://bin/libgdexample.dylib"
+    linux.debug = "res://bin/libgdexample.so"
+    windows.debug = "res://bin/libgdexample.dll"
+
+As you can see, ``gdexample_library_init`` is the same name of the function we
+defined in our ``init.c`` file. It is important that the names match because it
+is how Godot calls the entry point of the extension.
+
+We also set the compatibility minimum to 4.2, since we are targeting this
+version. It should still work on later versions. If you are using a later Godot
+version and rely on the new features, you need to increase this value to a
+version number that has everything you use.
```

Comment:
Since this page will be added in version 4.4, it may be worth explaining a little bit more why 4.2 is chosen as the target. Maybe here or maybe also at the start of the page, where the requirement of a 4.2+ Godot engine version is mentioned.

But maybe that is already covered in previous GDExtension pages.

Reply:
Generally, GDExtensions will target the lowest version that will work, in order to avoid having to making lots of different builds for different Godot versions, because a GDExtension built for an older version should work in a newer version (where as one built for a newer version won't work in an older version).

I don't see anywhere in the docs where we already talk about that, but I'm not sure this is the right place to go into it?

Reply:
Maybe another paragraph or sub-header under [Version Compatibility](https://docs.godotengine.org/en/latest/tutorials/scripting/gdextension/what_is_gdextension.html#version-compatibility), which can then be linked from other places as needed?

Reply:
Ah, that "Version Compatibility" section could certainly be improved! There's a warning further up on that same page which also say the thing about work on new versions but not older. Perhaps that warning should be removed and a similar statement put in "Version Compatibility"? Probably best in a separate PR, though

Reply:
Yeah, that change makes sense to defer for a later PR.

It's also been a couple minor versions so maybe the compatibility policy is a bit more solid now?

Reply:
Here's a PR that attempts it: https://github.com/godotengine/godot-docs/pull/10404

Reply:
> It's also been a couple minor versions so maybe the compatibility policy is a bit more solid now?

Our messaging has always been a little overly paranoid. :-) Except for the break between 4.0 and 4.1, we've really maintained compatibility in 99.99% of cases. We probably should soften the language about this, but there's some debate about how exactly - see https://github.com/godotengine/godot-cpp/pull/1588

Reply:
Yeah, that makes sense to me.

With https://github.com/godotengine/godot-docs/pull/10404 split out I think we can consider this resolved. If the other PR ends up merged before this one is merged, then maybe we should link to the explanation, but also maybe not since it appears in the intro page to GDExtension.

Reply:
Seems fine.

It is _slightly_ strange that the opening mentions 4.2, then it takes half the article to explain why we use 4.2. If there is some natural way to fit this into the introduction, that would be nice. But I'm not sure if there is a natural way, and this is an advanced tutorial, so the current approach seems totally fine to me.